frontend/src/app/chatmenu/BlockUser.tsx

```diff
@@ -1,66 +1,79 @@
-import React from 'react'
+import React, { useState } from 'react'
 import './BlockUser.css'
-import '../ui/User.css'
 
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { faUnlock, faUserLock } from '@fortawesome/free-solid-svg-icons'
+import { faLock, faLockOpen } from '@fortawesome/free-solid-svg-icons'
 
 import { ChatMenuPageProps } from './ChatMenuPage'
 import { UserData } from '../types/UserData'
+import { Button, UserTile } from '../ui'
+import { fetchFromBackend, useBackendFetchJson } from '../utils'
 
 interface BlockUserProps {
     participants: UserData[]
+    myUid: string
+}
+
+function FetchBlockedUsers(myUid: string, blockUid: string) {
+    const [res] = useBackendFetchJson({
+        route: `/chat/block/user/${myUid}`,
+        deps: [myUid],
+        isValid: myUid,
+    })
+    const blockedUsers = res?.blocks || []
+
+    return blockedUsers.find((user: { uid: string }) => user?.uid === blockUid)?.blocked || false
 }
 
 // create user tile to display participants
-const UserTile = ({ user }) => {
-    const { uid, displayName, photoURL } = user
-    const alt = `Profile image for ${displayName}.`
+const CreateUserTile = ({ user, myUid }) => {
+    const blockUid = user?.uid
+    const [isActive, setActive] = useState(FetchBlockedUsers(myUid, blockUid))
 
-    const doBlock = () => {
-        // TODO: Implement blocking the user method
+    const doBlock = async () => {
+        await fetchFromBackend({
+            route: `/chat/block/user/${myUid}/${blockUid}/true`,
+            options: { method: 'POST' },
+        })
     }
 
-    const doUnblock = () => {
-        // TODO: Implement unblocking the user method
+    const doUnblock = async () => {
+        await fetchFromBackend({
+            route: `/chat/block/user/${myUid}/${blockUid}/false`,
+            options: { method: 'POST' },
+        })
```

Comment:
These functions can also be partially lifted out of the `CreateUserTile` functional component. Then, the user tile doesn't need to have the extra prop of `myUid`.

Instead, the user tile can take a function prop:
```ts
doBlock(blockUid: string, shouldBlock: bool) => void
```

And then the component can use that function as the onclick callback. The parent component will take the input values and use that to send the actual request.

This also separates the component from the request concern, leaving only the event handling.